### PR TITLE
Add `relative-url-style` rule

### LINKED ISSUE #1614: Rule proposal: No moot `./` in `new URL()`
### Description

`new URL()` accepts relative URLs in the form `new URL('foo', url)` and `new URL('./foo', url)`. I would like to enforce the former, as adding the `./` is moot.

### Fail

```js
new URL('./foo', 'https://sindresorhus.com');
```

### Pass

```js
new URL('foo', 'https://sindresorhus.com');
```

### PR DESCRIPTION
**Pull request by fisker**
Fixes #1614

### CODE REVIEW

File: rules/relative-url-style.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
+'use strict';
+const {newExpressionSelector} = require('./selectors/index.js');
+const {replaceStringLiteral} = require('./fix/index.js');
+
+const MESSAGE_ID_NEVER = 'never';
+const MESSAGE_ID_ALWAYS = 'always';
+const messages = {
+	[MESSAGE_ID_NEVER]: 'Remove `./` prefix from the relative url.',
```
Comment (sindresorhus) [inline rules/relative-url-style.js:8]:
```suggestion
	[MESSAGE_ID_NEVER]: 'Remove the `./` prefix from the relative URL.',
```

File: rules/relative-url-style.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	},
+];
+
+/** @type {import('eslint').Rule.RuleModule} */
+module.exports = {
+	create,
+	meta: {
+		type: 'suggestion',
+		docs: {
+			description: 'Enforce consistent relative url style.',
```
Comment (sindresorhus) [inline rules/relative-url-style.js:97]:
```suggestion
			description: 'Enforce consistent relative URL style.',
```

File: rules/relative-url-style.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
+'use strict';
+const {newExpressionSelector} = require('./selectors/index.js');
+const {replaceStringLiteral} = require('./fix/index.js');
+
+const MESSAGE_ID_NEVER = 'never';
+const MESSAGE_ID_ALWAYS = 'always';
+const messages = {
+	[MESSAGE_ID_NEVER]: 'Remove `./` prefix from the relative url.',
+	[MESSAGE_ID_ALWAYS]: 'Use `./` prefix in the relative url.',
```
Comment (sindresorhus) [inline rules/relative-url-style.js:9]:
```suggestion
	[MESSAGE_ID_ALWAYS]: 'Add a `./` prefix to the relative URL.',
```

?

File: readme.md [hunk 4]
```diff
@@ -242,6 +243,7 @@ Each rule has emojis denoting:
 | [prefer-top-level-await](docs/rules/prefer-top-level-await.md) | Prefer top-level await over top-level promises and async function calls. |  |  | 💡 |
 | [prefer-type-error](docs/rules/prefer-type-error.md) | Enforce throwing `TypeError` in type checking conditions. | ✅ | 🔧 |  |
 | [prevent-abbreviations](docs/rules/prevent-abbreviations.md) | Prevent abbreviations. | ✅ | 🔧 |  |
+| [relative-url-style](docs/rules/relative-url-style.md) | Enforce consistent relative url style. | ✅ | 🔧 |  |
```
Comment (sindresorhus) [inline readme.md:246]:
```suggestion
| [relative-url-style](docs/rules/relative-url-style.md) | Enforce consistent relative URL style. | ✅ | 🔧 |  |
```

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: docs/rules/relative-url-style.md [hunk 5]
```diff
@@ -24,9 +24,9 @@ Type: `string`\
 Default: `'never'`
 
 - `'never'` (default)
-	- Never use a `./` prefix.
+  - Never use a `./` prefix.
 - `'always'`
-	- Always add a `./` prefix to the relative URL when possible.
+  - Always add a `./` prefix to the relative URL when possible.
```
Comment (fisker) [inline docs/rules/relative-url-style.md:29]: `markdownlint` enforce to use space.

**Pull request merged**